src/main/java/com/echobox/api/linkedin/connection/versioned/VersionedVideoConnection.java

```diff
@@ -34,13 +59,104 @@ public class VersionedVideoConnection extends VersionedConnection {
+    InitializeUploadResponse initializeUploadResponse = initializeUpload(initializeUploadRequest);
+    InitializeUploadResponse.Value value = initializeUploadResponse.getValue();
+    
+    List<String> uploadedPartIds = uploadVideoFile(filePath, value.getUploadInstructions());
+    
+    FinalizeUploadRequest finalizeUploadRequest =
+        new FinalizeUploadRequest(value.getVideo(), value.getUploadToken(), uploadedPartIds);
+    finalizeUpload(finalizeUploadRequest);
+    
+    return value.getVideo();
+  }
+  
+  public InitializeUploadResponse initializeUpload(
+      InitializeUploadRequest initializeUploadRequest) {
+    return linkedinClient.publish(VIDEOS, InitializeUploadResponse.class, initializeUploadRequest,
+        Parameter.with(ACTION_KEY, INITIALIZE_UPLOAD));
+  }
+  
+  public List<String> uploadVideoFile(String filePath,
+      List<InitializeUploadResponse.UploadInstruction> uploadInstructions) throws IOException {
+    
+    File file = new File(filePath);
+    byte[] fileBytes = convertToBytes(file);
+    
+    List<String> uploadPartIds = new ArrayList<>();
+    for (InitializeUploadResponse.UploadInstruction instruction : uploadInstructions) {
+      String etag = uploadVideoFileChunk(filePath, fileBytes, instruction);
+      uploadPartIds.add(etag);
+    }
+    
+    return uploadPartIds;
+  }
+  
+  public String uploadVideoFileChunk(String filePath, byte[] fileBytes,
+      InitializeUploadResponse.UploadInstruction instruction) throws IOException {
+    WebRequestor webRequestor = linkedinClient.getWebRequestor();
+    
+    Map<String, String> requestHeaders = new HashMap<>();
+    requestHeaders.put(DefaultVersionedLinkedInClient.HEADER_NAME_VERSION,
+        linkedinClient.getVersionedMonth());
+    
+    byte[] chunkBytes = Arrays.copyOfRange(fileBytes,
+        Math.toIntExact(instruction.getFirstByte()),
+        Math.toIntExact(instruction.getLastByte() + 1));
+    BinaryAttachment attachment = BinaryAttachment.with(filePath, chunkBytes,
+        ContentType.APPLICATION_OCTET_STREAM.toString());
+    
+    URL url = extractUploadURL(instruction.getUploadUrl());
+    WebRequestor.Response response =
+        webRequestor.executePut(url.toString(), null, null, requestHeaders, attachment);
+    Map<String, String> responseHeaders = response.getHeaders();
+    ValidationUtils.validateRequiredResponseHeader(responseHeaders, HEADER_ETAG);
+    
+    return responseHeaders.get(HEADER_ETAG);
+  }
+  
+  private URL extractUploadURL(String url) {
+    try {
+      return new URL(url);
+    } catch (MalformedURLException e) {
+      throw new LinkedInResponseException("Invalid upload url returned from LinkedIn.", e);
+    }
+  }
+  
+  public void finalizeUpload(FinalizeUploadRequest finalizeUploadRequest) {
+    linkedinClient.publish(VIDEOS, finalizeUploadRequest,
+        Parameter.with(ACTION_KEY, FINALIZE_UPLOAD));
+  }
+  
+  private static byte[] convertToBytes(File file) throws IOException {
```

Comment:
Just a note.
The max length of array is Integer.MAX. That's why all array-related functions used int as params.
We may need to read from the file chunk by chunk, instead of reading the whole file into an array.